From 0fd4fc8f4e8f51eec9c971c1170620dd64262bea Mon Sep 17 00:00:00 2001
Message-Id: <0fd4fc8f4e8f51eec9c971c1170620dd64262bea.1571399685.git.pmatilai@redhat.com>
From: Peter Robinson <pbrobinson@gmail.com>
Date: Wed, 16 Oct 2019 10:48:12 +0100
Subject: [PATCH] Remove problematic sub variants of armv8 and related

This is a partial revert of b1f81b837f46

The ARMv8 aarch32 variant specifies a number of required and optional
components. NEON is a required component of ARMv8 devices so we don't
need the 'n' variant that was added in ARMv7 (and never widely used due
to problems). The 'c' crypto variant shouldn't be added. The use of
the crypto extensions to by ARMv8 HW is many and varied and there's no
means of dealing with it at a package/compile extention so the detection
and use of it is done at runtime and handled via a number of different
mechanisms whether in code or via kernel crypto modules and the use of
libkcapi allowing the device to choose the fastest crypto option for
the use case.

This is actively breaking Fedora and related distributions on newer
generations of hardware actively stopping the actual install of an OS
in some extreme cases. This architecture functionality needs to be
reviewed by the architecture maintainers to ensure that active breakage
doesn't happen.

Fixes RHBZ: https://bugzilla.redhat.com/show_bug.cgi?id=1691430

Signed-off-by: Peter Robinson <pbrobinson@gmail.com>
---
 rpmrc.in | 9 ---------
 1 file changed, 9 deletions(-)

diff --git a/rpmrc.in b/rpmrc.in
index cc212d4ba..5bd9ba3e5 100644
--- a/rpmrc.in
+++ b/rpmrc.in
@@ -82,8 +82,6 @@ optflags: armv7hl -O2 -g -march=armv7-a -mfloat-abi=hard -mfpu=vfpv3-d16
 optflags: armv7hnl -O2 -g -march=armv7-a -mfloat-abi=hard -mfpu=neon
 optflags: armv8l -O2 -g -march=armv8-a
 optflags: armv8hl -O2 -g -march=armv8-a -mfloat-abi=hard -mfpu=vfpv4
-optflags: armv8hnl -O2 -g -march=armv8-a -mfloat-abi=hard -mfpu=neon-fp-armv8
-optflags: armv8hcnl -O2 -g -march=armv8-a -mfloat-abi=hard -mfpu=crypto-neon-fp-armv8
 
 optflags: m68k -O2 -g -fomit-frame-pointer
 
@@ -224,8 +222,6 @@ arch_canon:     armv7hl: armv7hl 	12
 arch_canon:     armv7hnl: armv7hnl 	12
 arch_canon:	armv8l: armv8l	12
 arch_canon:	armv8hl: armv8hl	12
-arch_canon:	armv8hnl: armv8hnl	12
-arch_canon:	armv8hcnl: armv8hcnl	12
 
 arch_canon:	m68kmint: m68kmint	13
 arch_canon:	atarist: m68kmint	13
@@ -353,8 +349,6 @@ buildarchtranslate: armv7hl: armv7hl
 buildarchtranslate: armv7hnl: armv7hnl
 buildarchtranslate: armv8l: armv8l
 buildarchtranslate: armv8hl: armv8hnl
-buildarchtranslate: armv8hnl: armv8hnl
-buildarchtranslate: armv8hcnl: armv8hcnl
 
 buildarchtranslate: mips: mips
 buildarchtranslate: mipsel: mipsel
@@ -471,9 +465,6 @@ arch_compat: armv5tl: armv4tl
 arch_compat: armv4tl: armv4l
 arch_compat: armv4l: armv3l
 arch_compat: armv3l: noarch
-arch_compat: armv8hcnl: armv8hnl
-arch_compat: armv8hnl: armv8hl
-arch_compat: armv8hnl: armv7hnl
 arch_compat: armv8hl: armv7hl
 arch_compat: armv7hnl: armv7hl
 arch_compat: armv7hl: armv6hl
-- 
2.21.0

